datalake/users_stack.py

```diff
@@ -0,0 +1,27 @@
+import os
+
+from aws_cdk import aws_iam, core
+from aws_cdk.core import Duration, Tags
+
+
+class UsersStack(core.Stack):
+    def __init__(self, scope: core.Construct, stack_id: str, **kwargs) -> None:
+        super().__init__(scope, stack_id, **kwargs)
+
+        principals = aws_iam.CompositePrincipal(
+            *(
```

Comment:
This is what I originally had, but I had a feeling that you wouldn't like it :) .Do you want me to change it ?

Reply:
Yes please :) I normally try to avoid nested function calls within the same line, mostly to make reading stack traces a bit more friendly. For comparison, in:

```python
foo(outer, *bar(middle, baz(inner)))
```

vs.

```python
bazzy = baz(inner)
barry = bar(middle, bazzy)
foo(outer, *barry)
```

I would almost always prefer the second version, since an error in any of the three calls will be trivial to locate in a stack trace and it's much easier to insert `import ipdb; ipdb.set_trace()` at exactly the right place.

Reply:
Done in 3bb3941c93fb90ec6ec3f23ff1b5fa527801ac31